test/integration/scheduler/predicates_test.go

```diff
@@ -1106,7 +1106,49 @@ func TestUnschedulablePodBecomesSchedulable(t *testing.T) {
 				return nil
 			},
 		},
-		// TODO(#91111): Add more test cases.
+		{
+			name: "other pod gets added",
```

Comment:
```suggestion
			name: "pod with pod-affinity gets added",
```

Reply:
Please, correct me if I am wrong, but I thought the purpose of https://github.com/kubernetes/kubernetes/issues/91111 is to add coverage to the events that triggers the scheduler. The tests you suggest seems to trigger the scheduler when a "pod gets deleted" which is covered already in https://github.com/kubernetes/kubernetes/pull/91734.

What makes you prefer these tests?

cc @alculquicondor

Reply:
Ok, let's back up a bit.

The idea of these tests is to cover the different events that change the state of the cluster. Ideally, we cover all variations of events that we could think of. But since these are integrations tests, and they are expensive, we should be a bit more conservative.

Then, we can cover the cases where we now a different path is taken. That is not the case for delete: all pods are moved back to the active queue. So I'm happy to just keep the test we have that uses the Fit plugin. The rest of the affinity workflow is already covered with unit tests.

wdyt @Huang-Wei

Reply:
@nodo @alculquicondor In terms of using integration tests to only cover different paths, I'm fine with holding off the extra deletion tests.

However, I think we missed the Update path, any thoughts on adding a test like this:

- init: create a node and a pod (without any label). The pod is expected to be scheduled.
- pod: create a pod without any lane, and with pod-affinity to co-exist with label "foo". The pod is expected to be pending - just like the current test.
- update: apply label "foo" to the scheduled pod. Then it's expected to trigger cache#UpdatePod(), and the pending pod is expected to be scheduled.

Reply:
Is that a valid case? Can you change labels in running Pods?
If so, I'm ok with having that test. Otherwise it's unnecessary.

Reply:
I tried both plain pod and podTemplate, they're both mutable.

Reply:
cool, it makes sense to me. I will add an additional test in this PR.

Reply:
Don't forget to update the name of the test to be more descriptive :)

`pod with pod-affinity gets added`

Reply:
aha! good point, thanks! :)